pkgs/development/python-modules/type-infer/default.nix

```diff
@@ -0,0 +1,74 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchPypi
+, poetry-core
+, colorlog
+, dataclasses-json
+, langid
+, nltk
+, numpy
+, pandas
+, psutil
+, python-dateutil
+, scipy
+, toml
+, nltk-data
+, symlinkJoin
+}:
+let
+  testNltkData = symlinkJoin {
+    name = "nltk-test-data";
+    paths = [ nltk-data.punkt nltk-data.stopwords ];
+  };
+in
+buildPythonPackage rec {
+  pname = "type-infer";
+  version = "0.0.15";
+  format = "pyproject";
+
+  disable = pythonOlder "3.8";
+
+  # using PyPI because the repo does not have tags or release branches
+  src = fetchPypi {
+    pname = "type_infer";
+    inherit version;
+    hash = "sha256-AnThYE6hHc3Pwu8fl0VBiQJfGVjeEKo4RrCsOl2pfCA=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    colorlog
+    dataclasses-json
+    langid
+    nltk
+    numpy
+    pandas
+    psutil
+    python-dateutil
+    scipy
+    toml
+  ];
+
+  buildEnv = with lib; [
+    "NLTK_DATA=${testNltkData}"
+  ];
+
+  # pythonImportsCheck is not used as the package requires NLTK_DATA to be be set, or it starts downloading data
+  # There is no way to set environment variables for pythonImportsCheck
+  # without tests I want to at least check that the package can be imported
```

Comment:
what happens if you set `env.NLTK_DATA` for the entire derivation, does it still fail ?

Reply:
also you can do an export in a preCheck phase if that doesn't work.

Reply:
setting env var in `preCheck` would not help, but `env.NLTK_DATA` works fine. Thank you! Fixed!